script.js

```diff
@@ -1,4 +1,76 @@
+//global variables
+var gameMode = "PLAYER ONE ROLL MODE";
+var playerOneDiceArray = [];
+var playerTwoDiceArray = [];
+var finalValuePlayerOne;
+var finalValuePlayerTwo;
+
+var rollDice = function () {
+  return Math.floor(Math.random() * 6) + 1;
+};
+
+var diceRoll1 = rollDice();
+var diceRoll2 = rollDice();
+var diceRoll3 = rollDice();
+var diceRoll4 = rollDice();
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  console.log(gameMode);
+  //PlayerOne
+  if (gameMode == "PLAYER ONE ROLL MODE") {
+    playerOneDiceArray.push(diceRoll1);
+    playerOneDiceArray.push(diceRoll2);
+
+    gameMode = "PLAYER ONE GUESS MODE";
+
+    return (myOutputValue = `Player One has rolled ${playerOneDiceArray[0]} and ${playerOneDiceArray[1]}. <br> To choose ${playerOneDiceArray[0]} as the first value, submit "1" . <br> To choose ${playerOneDiceArray[1]} as the first value, submit "2".`);
+  }
+
+  if (gameMode == "PLAYER ONE GUESS MODE") {
+    var firstValue = input;
+    var myOutputValue = ``;
+    if (firstValue == "1") {
+      finalValuePlayerOne = `${playerOneDiceArray[0]}${playerOneDiceArray[1]}`;
+      myOutputValue += `Player One's final value is ${finalValuePlayerOne}.`;
+    } else if (firstValue == "2") {
+      finalValuePlayerOne = `${playerOneDiceArray[1]}${playerOneDiceArray[0]}`;
+      myOutputValue += `Player One's final value is ${finalValuePlayerOne}.`;
+    }
+    gameMode = "PLAYER TWO ROLL MODE";
+    return myOutputValue;
+  }
```

Comment:
You'll notice that in the the GUESS MODE gameMode. Player 1 and 2 's code are almost identical. 
This is usually  a signal to decompose the logic here into a separate function that can be used for player1 and player2. 

Take a look at the reference solution for this exercise and how multiple helper functions are used 
[reference](https://github.com/rocketacademy/basics-beat-that/pull/10/files)